notebooks/Gradient Boosted Regression Trees - All Targets.ipynb

```diff
@@ -0,0 +1,230 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Gradient Boosting Regression - All targets\n",
+    "With single target GBR under our belts (see [Single target notebook](file/../Gradient Boosted Regression Trees - Single Target.ipynb)), \n",
+    "we can now evalute performance for all 5 targets by performing GBR\n",
+    "on each target separately.  We will end up with a model for each target.\n",
+    "\n",
+    "If you wish to follow the same method of selecting feature importances\n",
+    "as in the single GBR notebook, this can be achieved by setting up an\n",
+    "skl [Pipeline](http://scikit-learn.org/stable/modules/generated/sklearn.pipeline.Pipeline.html#sklearn.pipeline.Pipeline). \n",
+    "A sequence of transformations is easily applied to multiple targets.\n",
+    "\n",
+    "For simplicity, this notebook takes the first 100 principal components\n",
+    "to reduce the number of features."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "import pandas as pd\n",
+    "import sklearn as skl\n",
+    "import numpy as np\n",
+    "\n",
+    "import seaborn as sns\n",
+    "sns.set(font_scale=2)\n",
+    "\n",
+    "import scripts.load_data as load\n",
+    "\n",
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import sklearn.preprocessing\n",
+    "import sklearn.decomposition\n",
+    "import sklearn.ensemble\n",
+    "import sklearn.model_selection\n",
+    "import sklearn.feature_selection\n",
+    "import sklearn.metrics\n",
+    "\n",
+    "from scipy.stats import randint as sp_randint\n",
+    "from scipy.stats import uniform as sp_uniform\n",
+    "\n",
+    "from time import time"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "X,y=load.load_training_spectra()\n",
+    "pca = skl.decomposition.PCA(n_components=100)\n",
+    "X_transformed = pca.fit_transform(X)\n",
+    "\n",
+    "X_train, X_test, y_train, y_test = skl.model_selection.train_test_split(X_transformed,y,test_size=0.2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Two functions for different scoring metrics are in the cell below.  \n",
+    "scoreGBR returns r-squared as calculated from all targets.  \n",
+    "MCRMSE returns the kaggle scoring metric: mean columnwise RMSE, \n",
+    "the average of the RMSE found for each target.  "
```

Comment:
You could remove "found" here.

It might also be a good idea to define "RMSE" somewhere.